phone-number/example.rs

```diff
@@ -10,19 +10,18 @@ pub fn number(s: &str) -> String {
             _   => None
         },
         _  => None
-    }.unwrap_or("0000000000".to_string())
-
+    }
 }
 
-pub fn area_code(s: &str) -> String {
-    let n = number(s);
-    n[..3].to_string()
+pub fn area_code(s: &str) -> Option<String> {
+    number(s).map(|n| n[..3].to_string())
 }
 
-pub fn pretty_print(s: &str) -> String {
-    let n = number(s);
-    format!("({area}) {prefix}-{exchange}",
-            area=&n[..3],
-            prefix=&n[3..6],
-            exchange=&n[6..])
+pub fn pretty_print(s: &str) -> Option<String> {
```

Comment:
Sounds reasonable. And gives a chance to use or_else on the option. I'll update the PR if there are no further comments. 

From:"Eduardo Bautista" <notifications@github.com>
Date:Sun, 2 Aug, 2015 at 7:27
Subject:Re: [xrust] Use rust style error handling in phone number (#44)

In phone-number/example.rs:

> } > > -pub fn pretty_print(s: &str) -> String { > - let n = number(s); > - format!("({area}) {prefix}-{exchange}", > - area=&n[..3], > - prefix=&n[3..6], > - exchange=&n[6..]) > +pub fn pretty_print(s: &str) -> Option<String> { 

I am thinking that maybe pretty_print should return just a string with "Invalid Phone Number" in the case of number returning None.

—
Reply to this email directly or view it on GitHub.￼